core/winnt/src/TWinNTSystem.cxx

```diff
@@ -2256,20 +2256,27 @@ const char *TWinNTSystem::TempDirectory() const
 /// Create a secure temporary file by appending a unique
 /// 6 letter string to base. The file will be created in
 /// a standard (system) directory or in the directory
-/// provided in dir. The full filename is returned in base
+/// provided in dir. Optionally one can provide suffix
+/// append to the final name - like extension ".txt" or ".html".
+/// The full filename is returned in base
 /// and a filepointer is returned for safely writing to the file
 /// (this avoids certain security problems). Returns 0 in case
 /// of error.
 
-FILE *TWinNTSystem::TempFileName(TString &base, const char *dir)
+FILE *TWinNTSystem::TempFileName(TString &base, const char *dir, const char *suffix)
 {
    char tmpName[MAX_PATH];
 
    ::GetTempFileName(dir ? dir : TempDirectory(), base.Data(), 0, tmpName);
    base = tmpName;
-   FILE *fp = fopen(tmpName, "w+");
+   if (suffix && *suffix) {
+      TString base_suffix = base + suffix;
```

Comment:
We should add a comment here. In theory, it is possible that the temporary file _with_ the suffix already exists. I don't think we can do better on Windows, since there is no `mkstemps`. But at least we should mention it.

There is a possible solution: create a temporary unique `directory` and in that a file with the wanted suffix. But I think that's overkill and also more annoying to clean up (remove file + directory).